matrix/matrix.py

```diff
@@ -0,0 +1,176 @@
+from typing import Union
+
+
+class Matrix:
+
+    def __init__(self, matrix: tuple):
+        Matrix.is_valid_matrix(matrix)
```

Comment:
Use "self" instead of Matrix (here and everywhere) it will be more generic. (Tomorrow we will change the name of Matrix to YoniBayoniMatrix, it will be boomer to change all to occurrences of Matrix explicitly)